### PR TITLE
Merging pubsub and rediscache

### PR DESCRIPTION
**Pull request by bloodbare**

### CODE REVIEW

File: guillotina/contrib/cache/__init__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from guillotina import configure
+
+
+CACHE_PREFIX = 'gcache2-'
+
+app_settings = {
+    'cache': {
+        'driver': 'guillotina.contrib.redis',  # Empty to use memory
```
Comment (vangheem) [inline guillotina/contrib/cache/__init__.py]: Should we have default be empty then?

File: guillotina/contrib/cache/__init__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        'updates_channel': 'guillotina',  # Empty to not subscribe
```
Comment (vangheem) [inline guillotina/contrib/cache/__init__.py]: same here?

File: guillotina/contrib/cache/utility.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+        assert isinstance(msg, dict)
+        assert 'tid' in msg
+        assert 'keys' in msg
+        if msg['tid'] in self._ignored_tids:
+            # on the same thread, ignore this sucker...
+            self._ignored_tids.remove(msg['tid'])
+            return
+
+        mem_cache_obj = memcache.get_memory_cache()
```
Comment (vangheem) [inline guillotina/contrib/cache/utility.py]: should we use `self._memory_cache` instead?

File: guillotina/permissions.py [hunk 4]
```diff
@@ -173,6 +174,9 @@
 configure.grant(
     permission="guillotina.RawSearchContent",
     role="guillotina.ContainerAdmin")
+configure.grant(
```
Comment (vangheem) [inline guillotina/permissions.py]: Should this be defined in `guillotina.contrib.cache`?

File: guillotina/contrib/pubsub/__init__.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+app_settings = {
```
Comment (vangheem) [inline guillotina/contrib/pubsub/__init__.py]: Should we add dependency of `guillotina.contrib.cache`?

`"applications": ["guillotina.contrib.cache"]`?

File: guillotina/contrib/pubsub/utility.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
+from guillotina.profile import profilable
+from guillotina.utils import resolve_dotted_name
+from guillotina.contrib.pubsub.exceptions import NoPubSubDriver
+from typing import Any
+from typing import Callable
+import pickle
+import asyncio
+import logging
+
+app_settings = {
```
Comment (vangheem) [inline guillotina/contrib/pubsub/utility.py]: Why are these app_settings here? Shouldn't it be loading from the app_settings in parent module?

File: guillotina/contrib/redis/__init__.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+from guillotina.contrib.redis.driver import RedisDriver
+from asyncio import get_running_loop
+_driver = None
+
+app_settings = {
```
Comment (vangheem) [inline guillotina/contrib/redis/__init__.py]: Same here about dependency maybe?

File: guillotina/contrib/cache/__init__.py (hunk 1 above)
Comment (bloodbare) [reply guillotina/contrib/cache/__init__.py]: Yups

File: guillotina/contrib/cache/__init__.py (hunk 2 above)
Comment (bloodbare) [reply guillotina/contrib/cache/__init__.py]: yups

File: guillotina/contrib/cache/utility.py (hunk 3 above)
Comment (bloodbare) [reply guillotina/contrib/cache/utility.py]: yups

**Pull request merged**